src/main/java/org/takes/facets/auth/codecs/CcAES.java

```diff
@@ -44,77 +43,56 @@
  * AES codec which supports 128 bits key.
  *
  * <p>The class is immutable and thread-safe.
- *
  * @author Jason Wong (super132j@yahoo.com)
  * @version $Id$
  * @since 0.13.8
  */
-@EqualsAndHashCode(of = { "origin", "secret", "enc", "dec" })
+@EqualsAndHashCode(of = {"origin", "key"})
 public final class CcAES implements Codec {
+    /**
+     * The block size constant.
+     */
+    private static final int BLOCK = 16;
 
     /**
-     * The cipher for encryption.
+     * Original codec.
      */
-    private final transient Cipher enc;
+    private final transient Codec origin;
 
     /**
-     * The cipher for decryption.
+     * The encryption key.
      */
-    private final transient Cipher dec;
+    private final transient byte[] key;
 
     /**
-     * Original codec.
+     * The algorithm parameter spec for cipher.
      */
-    private final transient Codec origin;
+    private transient AlgorithmParameterSpec spec;
 
     /**
-     * The AES secret key object.
+     * Tracks the spec variable is initialized or not.
      */
-    private final transient SecretKey secret;
+    private transient boolean initialized;
 
     /**
-     * Ctor.
+     * Constructor for the class.
      * @param codec Original codec
      * @param key The encryption key
-     * @exception IOException errors on creating internal components
      * @since 0.22
      */
-    public CcAES(final Codec codec, final String key) throws IOException {
-        this(codec, key.getBytes(Charset.defaultCharset()));
+    public CcAES(final Codec codec, final String key) {
+        this(codec, key.getBytes(StandardCharsets.UTF_8));
```

Comment:
@mkordas When i run them local i am not getting the error but in Travis I am getting this error
```
[WARNING] FindBugs: H I Dm: Found reliance on default encoding in new org.takes.facets.auth.codecs.CcAES(Codec, String): String.getBytes()  At CcAES.java:[line 79] 
[INFO] Read our quality policy: http://www.qulice.com/quality.html 
[INFO] ------------------------------------------------------------------------ 
[INFO] BUILD FAILURE 
```
So I am adding them back. Sorry for the confusion.